go/enclave/components/batch_executor.go

```diff
@@ -344,8 +356,23 @@ func (executor *batchExecutor) execMempoolTransactions(ec *BatchExecutionContext
 		if ltx == nil {
 			break
 		}
+
+		gasCost, err := executor.gasOracle.EstimateL1StorageGasCost(ec.ctx, ltx.Tx, ec.l1block, ec.currentBatch.Header)
```

Comment:
why not call oracle.calculateL1Cost here? Seems to be duplicating that logic

Reply:
It's a convinience wrapper that also handles rlp encoding, which is something the oracle should know how to deal with given a transaction or a message, thus why the calculateL1Cost is wrapped in them